crit/stats.go

```diff
@@ -0,0 +1,38 @@
+package crit
+
+import (
+	"errors"
+	"path/filepath"
+
+	"github.com/checkpoint-restore/go-criu/v6/crit/images"
+)
+
+func GetDumpStats(dir string) (*images.DumpStatsEntry, error) {
+	c := New(filepath.Join(dir, "stats-dump"), "", "", false, false)
+	statsImg, err := c.Decode()
+	if err != nil {
+		return nil, err
+	}
+
+	stats, ok := statsImg.Entries[0].Message.(*images.StatsEntry)
+	if !ok {
+		return nil, errors.New("Failed to type assert stats-dump image")
+	}
+
+	return stats.GetDump(), nil
+}
+
+func GetRestoreStats(dir string) (*images.RestoreStatsEntry, error) {
+	c := New(filepath.Join(dir, "stats-restore"), "", "", false, false)
+	statsImg, err := c.Decode()
+	if err != nil {
+		return nil, err
+	}
+
+	stats, ok := statsImg.Entries[0].Message.(*images.StatsEntry)
+	if !ok {
+		return nil, errors.New("Failed to type assert stats-restore image")
+	}
```

Comment:
The code in these two functions looks very similar.
Would it be possible to move it into `getStats()` function that is then used by `GetDumpStats()` and `GetRestoreStats()`?

Reply:
The file name changes for both functions (stats-dump and stats-restore). I feel like reworking it into a common function will actually make it worse to implement

Reply:
What about something like the following?
```go
func getStats(path string) (*images.StatsEntry, error) {
	c := New(path, "", "", false, false)
	statsImg, err := c.Decode()
	if err != nil {
		return nil, err
	}

	stats, ok := statsImg.Entries[0].Message.(*images.StatsEntry)
	if !ok {
		return nil, errors.New("Failed to type assert stats image")
	}

	return stats, nil
}

func GetDumpStats(dir string) (*images.DumpStatsEntry, error) {
	stats, err := getStats(filepath.Join(dir, "stats-dump"))
	if err != nil {
		return nil, err
	}

	return stats.GetDump(), nil
}

func GetRestoreStats(dir string) (*images.RestoreStatsEntry, error) {
	stats, err := getStats(filepath.Join(dir, "stats-restore"))
	if err != nil {
		return nil, err
	}

	return stats.GetRestore(), nil
}
```

Reply:
Yep I landed at the same thing :+1: